decayAmplitude/diffractiveDissVertex.cc

```diff
@@ -336,3 +338,10 @@ diffractiveDissVertex::printPointers(ostream& out) const
 	    << "recoil particle = " << recoil()    << endl;
 	return out;
 }
+
+double diffractiveDissVertex::getTprime() const{
+	const TLorentzVector lvBeam = TLorentzVector(beam()->lzVec());
```

Comment:
Why are you using the copy constructors here? Also, why are you not using references?